sklearn/model_selection/_prediction.py

```diff
@@ -0,0 +1,801 @@
+        the entire set will be used.
+
+    scorer : scorer instance
+        The scorer taking `classifier` and the validation set as input and outputting
+        decision thresholds and scores.
+
+    score_method : str or callable
+        The scoring method to use. Used to detect if we compute TPR/TNR or precision/
+        recall.
+
+    Returns
+    -------
+    thresholds : ndarray of shape (n_thresholds,)
+        The decision thresholds used to compute the scores. They are returned in
+        ascending order.
+
+    scores : ndarray of shape (n_thresholds,) or tuple os such arrays
+        The scores computed for each decision threshold. When TPR/TNR or precision/
+        recall are computed, `scores` is a tuple of two arrays.
+    """
+    arrays = (X, y) if sample_weight is None else (X, y, sample_weight)
+    check_consistent_length(*arrays)
+
+    fit_parameters = signature(classifier.fit).parameters
+    supports_sw = "sample_weight" in fit_parameters
+
+    if train_idx is not None:
+        X_train, X_val = _safe_indexing(X, train_idx), _safe_indexing(X, val_idx)
+        y_train, y_val = _safe_indexing(y, train_idx), _safe_indexing(y, val_idx)
+        if sample_weight is not None:
+            sw_train, sw_val = (
+                _safe_indexing(sample_weight, train_idx),
+                _safe_indexing(sample_weight, val_idx),
+            )
+        else:
+            sw_train, sw_val = None, None
+        fit_params_train = _check_fit_params(X, fit_params, indices=train_idx)
+        if supports_sw:
+            classifier.fit(X_train, y_train, sample_weight=sw_train, **fit_params_train)
+        else:
+            classifier.fit(X_train, y_train, **fit_params_train)
+    else:  # prefit estimator, only a validation set is provided
+        X_val, y_val, sw_val = X, y, sample_weight
+        check_is_fitted(classifier, "classes_")
+
+    if isinstance(score_method, str):
+        if score_method in {"max_tpr_at_tnr_constraint", "max_tnr_at_tpr_constraint"}:
+            fpr, tpr, potential_thresholds = scorer(
+                classifier, X_val, y_val, sample_weight=sw_val
+            )
+            # thresholds are in decreasing order
+            return potential_thresholds[::-1], ((1 - fpr)[::-1], tpr[::-1])
+        elif score_method in {
+            "max_precision_at_recall_constraint",
+            "max_recall_at_precision_constraint",
+        }:
+            precision, recall, potential_thresholds = scorer(
+                classifier, X_val, y_val, sample_weight=sw_val
+            )
+            # thresholds are in increasing order
+            # the last element of the precision and recall is not associated with any
+            # threshold and should be discarded
+            return potential_thresholds, (precision[:-1], recall[:-1])
+    return scorer(classifier, X_val, y_val, sample_weight=sw_val)
+
+
+class CutOffClassifier(ClassifierMixin, MetaEstimatorMixin, BaseEstimator):
```

Comment:
About the name of the meta estimator, I think I would be in favor of `ClassificationThresholdTuner` or `TunedThresholdClassifier`.